Experiment_Launcher_code/ExperimentLauncher.py

```diff
@@ -6,27 +15,34 @@ def main():
     experiment_gui = ExperimentGUI()
     experiment_gui.setup_gui()
 
-    # After the GUI is closed, get the settings using the get_settings method
-    settings = experiment_gui.on_start_clicked()
-
-    if settings:
-        # Extract the necessary parameters from the settings
-        experiment_name = settings.get('experiment_name')
-        comport_name = settings.get('comport_name')
-        num_trials = settings.get('num_trials')
-        return_time = settings.get('return_time')
-        decision_time = settings.get('decision_time')
-        opponent_type = settings.get('opponent_type')
-        opponent1_strategy = settings.get('opponent1_strategy')
-        opponent2_strategy = settings.get('opponent2_strategy')
-        opponent1_probability = settings.get('opponent1_probability')
-        opponent2_probability = settings.get('opponent2_probability')
+    # After the GUI is closed, get the settings using the appropriate methods
+    if experiment_gui.is_data_valid():
```

Comment:
why not check validity before passing to thr experimentlauncher in the gui program itself?